plonkish_backend/src/accumulation/protostar/ivc/aggregation/agg_circuit.rs

```diff
@@ -154,153 +152,139 @@ impl CircuitExt<grumpkin::Fr> for SecondaryAggregationCircuit {
+            self.vp_digest,
+            self.vp.clone(),
+            self.primary_arity,
+            chip.clone(),
+            chip.clone(),
+        );
+
+        let mut transcript = strawman::PoseidonTranscriptChip::new(
+            builder.main(0),
+            strawman::decider_transcript_param(),
+            chip.clone(),
+            self.proof.clone(),
+        );
+
+        let (primary_num_steps, primary_initial_input, primary_output, h_ohs_from_last_nark) =
+            aggregator.verify_ipa_grumpkin_ivc_with_last_nark(
+                &mut pool,
+                self.num_steps,
+                self.initial_input.clone(),
+                self.output.clone(),
+                self.acc_before_last.clone(),
+                self.last_instance,
+                &mut transcript,
+            )?;
+
+        let (secondary_initial_input, secondary_output, pairing_acc) = {
+            let mut transcript = strawman::PoseidonTranscriptChip::new(
+                builder.main(0),
+                strawman::decider_transcript_param(),
+                chip.clone(),
+                self.secondary_aggregation_proof.clone(),
+            );
+
+            let secondary_aggregation_instance = chip.verify_hyrax_hyperplonk(
+                &mut pool,
+                &self.secondary_aggregation_vp,
+                self.secondary_aggregation_instances
+                    .as_ref()
+                    .map(Vec::as_slice),
+                &mut transcript,
+            )?;        
+            let secondary_num_steps = chip.fit_base_in_scalar(&mut pool, &secondary_aggregation_instance[0])?;
+            chip.constrain_equal(&mut pool, &primary_num_steps, &secondary_num_steps)?;
+
+            let h = chip.fit_base_in_scalar(&mut pool, &secondary_aggregation_instance[1 + 2 * self.secondary_arity],
+            )?;
+            chip.constrain_equal(&mut pool, &h_ohs_from_last_nark, &h)?;
+
+            let iter = &mut secondary_aggregation_instance.iter();
+            let mut instances = |skip: usize, take: usize| {
+                iter.skip(skip)
+                    .take(take)
+                    .map(|base| chip.to_repr_base(base))
+                    .try_collect::<_, Vec<_>, _>()
+            };
+            (
+                instances(1, self.secondary_arity)?,
+                instances(0, self.secondary_arity)?,
+                instances(1, 4 * strawman::NUM_LIMBS)?,
+            )
+        };
+
+        // let cell_map = chip.layout_and_clear(&mut layouter)?;
+        // for (idx, witness) in chain![
+        //     [primary_num_steps],
+        //     primary_initial_input,
+        //     primary_output,
+        //     secondary_initial_input.into_iter().flatten(),
+        //     secondary_output.into_iter().flatten(),
+        //     pairing_acc.into_iter().flatten(),
+        // ]
+        // .enumerate()
+        // {
+        //     layouter.constrain_instance(cell_map[&witness.id()].cell(), chip.instance, idx)?;
```

Comment:
layouter.constrain_instance is exposing these values as public inputs but we dont use this api. axiom has a different api which you can find in synthesize_accumulation_verifier